tests/components/metoffice/test_metoffice.py

```diff
@@ -0,0 +1,314 @@
+"""The tests for the Met Office weather component."""
+from datetime import datetime, timedelta
+import json
+
+from homeassistant.components.metoffice.const import ATTRIBUTION, DOMAIN
+from homeassistant.const import STATE_UNAVAILABLE
+from homeassistant.util import utcnow
+
+from .const import (
+    DATETIME_FORMAT,
+    METOFFICE_CONFIG_KINGSLYNN,
+    METOFFICE_CONFIG_WAVERTREE,
+    TEST_DATETIME_STRING,
+    TEST_SITE_NAME_KINGSLYNN,
+    TEST_SITE_NAME_WAVERTREE,
+)
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry, async_fire_time_changed, load_fixture
+
+KINGSLYNN_SENSOR_RESULTS = {
+    "weather": ("weather", "sunny"),
+    "visibility": ("visibility", "Very Good"),
+    "visibility_distance": ("visibility_distance", "20-40"),
+    "temperature": ("temperature", "14"),
+    "feels_like_temperature": ("feels_like_temperature", "13"),
+    "uv": ("uv_index", "6"),
+    "precipitation": ("probability_of_precipitation", "0"),
+    "wind_direction": ("wind_direction", "E"),
+    "wind_gust": ("wind_gust", "7"),
+    "wind_speed": ("wind_speed", "2"),
+    "humidity": ("humidity", "60"),
+}
+
+WAVERTREE_SENSOR_RESULTS = {
+    "weather": ("weather", "sunny"),
+    "visibility": ("visibility", "Good"),
+    "visibility_distance": ("visibility_distance", "10-20"),
+    "temperature": ("temperature", "17"),
+    "feels_like_temperature": ("feels_like_temperature", "14"),
+    "uv": ("uv_index", "5"),
+    "precipitation": ("probability_of_precipitation", "0"),
+    "wind_direction": ("wind_direction", "SSE"),
+    "wind_gust": ("wind_gust", "16"),
+    "wind_speed": ("wind_speed", "9"),
+    "humidity": ("humidity", "50"),
+}
+
+
+class MockDateTime(datetime):
+    """Replacement for datetime that can be mocked for testing."""
+
+    def __new__(cls, *args, **kwargs):
+        """Override to just return base class."""
+        return datetime.__new__(datetime, *args, **kwargs)
+
+
+@patch("datetime.datetime", MockDateTime)
```

Comment:
I think we should try to patch something like this instead, but set the patch target path to the library path for datetime.now:
https://github.com/home-assistant/core/blob/3c57475c8f308231c001e816b105d6739a72868c/tests/components/caldav/test_calendar.py#L335

Reply:
Sadly, this doesn't work - if you see in [this pastebin of the test file](https://pastebin.com/WDQm62Gg) I've tried the functions in various combinations of patches, but none work (unless I've missed your point) as you can see [from the output](https://pastebin.com/mrYXakUP).

Reply:
Eureka! Thanks for the assist - did a bit more digging and reading, and found where to patch the datetime override. I'll check it in and hopefully it'll be more what you were looking for :)

Reply:
Good! I tested this on my end too and this works:
```suggestion
@patch(
    "datapoint.Forecast.datetime.datetime",
    Mock(now=Mock(return_value=datetime(2020, 4, 25, 12, tzinfo=timezone.utc))),
)
```

Reply:
Oh I like how you've got the mock return type in there (I've so much to learn, still...) - I'd not managed to get it that neat. I'll update and re-commit.